cavern/src/main/java/org/opencadc/cavern/files/PutAction.java

```diff
@@ -89,26 +89,27 @@
 import java.security.AccessControlException;
 import java.security.DigestInputStream;
 import java.security.MessageDigest;
-import java.util.Iterator;
 import org.apache.log4j.Logger;
 import org.opencadc.cavern.nodes.NodeUtil;
 
 /**
  *
  * @author majorb
+ * @author jeevesh
  */
-public class PutAction extends FileAction {
+public abstract class PutAction extends FileAction {
     private static final Logger log = Logger.getLogger(PutAction.class);
 
     private static final String INPUT_STREAM = "in";
 
-    public PutAction() {
-        super();
+    public PutAction(Direction pushToVoSpace, boolean isPreauth) {
+        super(pushToVoSpace, isPreauth);
     }
 
     @Override
-    public Direction getDirection() {
-        return Direction.pushToVoSpace;
+    public void initAction() throws Exception {
+        // Authorization is checked here
+        initNodeURI(syncInput.getPath());
     }
```

Comment:
since initAction() is the same in GetAction and PutAction, can the callling of that just be done automatically in FileAction without the subclasses having to initiate it?

Reply:
Had to test that it worked. Seems to. I'd tried putting the init for nodeURI in a few places before I found one (this) where syncInput was available.